cockatrice/src/dlg_update.cpp

```diff
@@ -0,0 +1,114 @@
+#define HUMAN_DOWNLOAD_URL "https://bintray.com/cockatrice/Cockatrice/Cockatrice/_latestVersion"
+
+#include <QtNetwork>
+#include <QProgressDialog>
+#include <QtGui/qdesktopservices.h>
+#include <QtWidgets/qmessagebox.h>
+
+#include "dlg_update.h"
+
+DlgUpdate::DlgUpdate(QWidget *parent, QString downloadUrlStr, int downloadSize)
+        : QProgressDialog(parent) {
+
+    QUrl downloadUrl = QUrl(downloadUrlStr);
+
+    //Check for SSL (this probably isn't necessary)
+    if (!QSslSocket::supportsSsl()) {
+        QMessageBox::critical(
+                this,
+                tr("Error"),
+                tr("Cockatrice was not built with SSL support, so cannot download update! "
+                           "Please visit download the new version manually at the following URL: " HUMAN_DOWNLOAD_URL));
+    }
+
+    //Update the maximum progress
+    setMaximum(downloadSize);
+
+    //Work out the file name of the download
+    QString fileName = downloadUrlStr.section('/', -1);
+
+    //Create a temporary directory to save the file in
+    QTemporaryDir tempDir;
+    tempDir.setAutoRemove(false);
+
+    //Open the file for writing and begin the download
+    file = new QFile(tempDir.path() + '/' + fileName);
+    if (file->open(QIODevice::WriteOnly))
+        beginDownload(downloadUrl);
+    else
+        QMessageBox::critical(
+                this,
+                tr("Error"),
+                tr("A temporary file could not be opened for downloading. Please update manually from " HUMAN_DOWNLOAD_URL));
+}
+
+void DlgUpdate::beginDownload(QUrl downloadUrl) {
+
+    //Request the file and handle success failure, progress etc.
+    response = netMan.get(QNetworkRequest(downloadUrl));
+    connect(response, SIGNAL(finished()),
+            this, SLOT(fileFinished()));
+    connect(response, SIGNAL(readyRead()),
+            this, SLOT(fileReadyRead()));
+    connect(response, SIGNAL(downloadProgress(qint64, qint64)),
+            this, SLOT(fileProgress(qint64, qint64)));
+    connect(response, SIGNAL(error(QNetworkReply::NetworkError)),
+            this, SLOT(downloadError(QNetworkReply::NetworkError)));
+}
+
+void DlgUpdate::downloadError(QNetworkReply::NetworkError) {
+    //Alert the user to the error then close
```

Comment:
This type of comment isn't useful, it just re-states what the code below it does.